src/main/java/com/exactpro/th2/lib/template/metric/RateHistogram.java

```diff
@@ -19,7 +43,33 @@ protected RateHistogram(ScheduledExecutorService scheduledExecutorService,
         this.scheduler = scheduledExecutorService.scheduleAtFixedRate(this::observeRates, intervalMs, intervalMs, TimeUnit.MILLISECONDS);
     }
 
-    protected abstract void observeRates();
+    protected void observeRates() {
```

Comment:
> protected

Not sure that it should be overridable

Reply:
It is impossible to specify generic type constraints in such a way that we can use both String and String[]. So implementations of RateHistogram have to provide method to convert generic `T` to String or String[] to pass to `histogram.labels()`. 
In this commit I extracted only necessary part of this logic, but it didn't solve the entire problem anyway

Reply:
My point is that there's no reason to override this method i.e. you can make it private